test/object-store/util/test_file.cpp

```diff
@@ -61,15 +61,7 @@ TestFile::TestFile()
 {
     static std::string tmpdir = [] {
         disable_sync_to_disk();
-
-        const char* dir = getenv("TMPDIR");
-        if (dir && *dir)
```

Comment:
Im not an expert on this, but it seems like we require this for linux

Reply:
The intention here was to have a single implementation. So we already have a function in test_utils.cpp called tmp_dir() that does the exact same thing as here. This is just consolidating that.